keptn/keptn
# feat: Introduce tolerations and affinities for helm charts

Signed-off-by: Philipp Hinteregger <philipp.hinteregger@dynatrace.com>




## This PR


- introduces tolerations for Helm charts
- introduces affinity for Helm charts
- introduces nodeAffinity for Helm charts
- introduces podAffinity for Helm charts
- introduces podAntiAffinity for Helm charts

If the affinity option is set in the values.yaml, the other affinity options will be ignored. 
Configurations made on the service level, will overwrite the default values.

### Related Issues


Fixes #8568

### How to test


- Set the desired values in the values.yaml
- Use `helm template --dry-run` to check if the generated output is valid
